[qt5-webengine] build failure after upgrading to ninja 1.12

### Operating system

Linux

### Compiler

GCC

### Steps to reproduce the behavior

```Shell
./vcpkg install --triplet=x64-linux-dynamic qt5-webengine
```

### Failure logs

```
In file included from ../../../../src/5.15.16-af4f9b3d26.clean/src/3rdparty/chromium/content/browser/appcache/appcache_internals_ui.cc:25:
../../../../src/5.15.16-af4f9b3d26.clean/src/3rdparty/chromium/content/public/browser/browser_context.h:21:10: fatal error: components/spellcheck/spellcheck_buildflags.h: No such file or directory
   21 | #include "components/spellcheck/spellcheck_buildflags.h"
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
```

### Additional context

I think it is related to this issue: https://github.com/ninja-build/ninja/issues/2417.

--- a/ports/qt5-webengine/fix-spellcheck-buildflags.patch
+++ b/ports/qt5-webengine/fix-spellcheck-buildflags.patch
@@ -1,0 +1,12 @@
+diff --git a/src/3rdparty/chromium/content/browser/BUILD.gn b/src/3rdparty/chromium/content/browser/BUILD.gn
+index 1466f33d4..3fc848a56 100644
+--- a/src/3rdparty/chromium/content/browser/BUILD.gn
++++ b/src/3rdparty/chromium/content/browser/BUILD.gn
+@@ -40,6 +40,7 @@ jumbo_static_library("devtools_protocol") {
+     "//content/browser/devtools:protocol_sources",
+     "//content/common:buildflags",
+     "//third_party/inspector_protocol:crdtp",
++    "//components/spellcheck:buildflags",
+   ]
+ 
+   sources = [

--- a/ports/qt5-webengine/portfile.cmake
+++ b/ports/qt5-webengine/portfile.cmake
@@ -64,6 +64,7 @@ set(PATCHES common.pri.patch
             0001-Support-ICU-74-in-LazyTextBreakIterator.patch
             workaround-protobuf-issue.patch
             0001-Fix-jumbo-build-error-due-to-ResolveColor-redefiniti.patch
+	    fix-spellcheck-buildflags.patch
             )
 
 set(OPTIONS)

--- a/ports/qt5-webengine/vcpkg.json
+++ b/ports/qt5-webengine/vcpkg.json
@@ -1,7 +1,7 @@
 {
   "name": "qt5-webengine",
   "version": "5.15.16",
-  "port-version": 1,
+  "port-version": 2,
   "description": "Qt WebEngine provides functionality for rendering regions of dynamic web content.",
   "license": null,
   "supports": "!static",

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -7538,7 +7538,7 @@
     },
     "qt5-webengine": {
       "baseline": "5.15.16",
-      "port-version": 1
+      "port-version": 2
     },
     "qt5-webglplugin": {
       "baseline": "5.15.16",

--- a/versions/q-/qt5-webengine.json
+++ b/versions/q-/qt5-webengine.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "7785e40d08d000118d180514d7e5f5f30a75ada8",
+      "version": "5.15.16",
+      "port-version": 2
+    },
+    {
       "git-tree": "92f56035f25cfc039d74c70b6905c1b78e480e2e",
       "version": "5.15.16",
       "port-version": 1

